test/TestProjects/Pagination-TypeSpec/Pagination-TypeSpec.tsp

```diff
@@ -212,18 +212,86 @@ model TextBlockItem {
   text: string;
 }
 
+enum Interval {
+  @doc("5 seconds, available only if test run duration is less than 10 minutes")
+  PT5S,
+  @doc("10 seconds, available only if test run duration is less than 10 minutes")
+  PT10S,
+  @doc("1 minute") PT1M,
+  @doc("5 minutes, available only if test run duration is greater than 1 minute")
+  PT5M,
+  @doc("1 hour, available only if test run duration is greater than 1 minute")
+  PT1H,
+}
+
+model MetricDimensions {
+  @doc("The interval (i.e. timegrain) of the query.")
+  @query
+  interval?: Interval;
+
+  @doc("Metric name")
+  @query
+  metricName?: string;
+
+  @doc("Metric namespace to query metric definitions for.")
+  @query
+  metricNamespace: string;
+
+  @doc("""
+  The timespan of the query. It is a string with the following format
+  'startDateTime_ISO/endDateTime_ISO'.
+  """)
+  @query
+  timespan?: string;
+};
+
+@resource("values")
+@parentResource(Dimension)
+model DimensionValueList {
+  @key
+  value: string[];
+}
+
+@resource("metric-dimensions")
+@parentResource(TestRuns)
+model Dimension {
+  @key
+  @doc("Dimension name")
+  @path("name")
+  @visibility("query")
+  name: string;
+}
+
+@doc("Load test run model")
+@resource("test-runs")
+model TestRuns {
+  // FIXME: is testRunId optional or required in the response? Resource key and path params cannot be optional
+  @key
+  @path
+  @doc("Unique test run name as identifier")
+  @visibility("read")
+  testRunId: string;
+}
+
 //
 // ** Operations **
 //
 
 @route("/app")
 namespace Pagination {
-
   @summary("Gets ledger entries from a collection corresponding to a range.")
   @doc("A collection id may optionally be specified. Only entries in the specified (or default) collection will be returned.")
   op ListPaginationLedgerEntries is Azure.Core.Foundations.ResourceList<LedgerEntry, ListLedgerEntryInput, PagedLedgerEntries>;
 }
 
+interface AdditionalParameter {  
+  @summary("List the dimension values for the given metric dimension name.")
+  @doc("List the dimension values for the given metric dimension name.")
+  ListMetricDimensionValues is Azure.Core.ResourceList<
+    DimensionValueList, QueryParametersTrait<MetricDimensions>
+  >;
```

Comment:
What does the first generic parameter here in `ResourceList` mean? Is it the type of the item or the type of the `XXXListResult` which represents a page?

We are generating a method returning `Pageable<DimensionValueList>`, therefore, if our pageable implementation is correct, the generic parameter above means the type of the element. If so, I suggest we change the model name here, or just use string, or add some comment here to describe that this model `DimensionValueList` is the type of the element of the pageable operation to avoid future confusion

Reply:
DimensionValueList is the item type. 

Emmmm, it comes from service definition. Maybe we should ask service team to change?